src/wms/src/main/java/org/geoserver/wms/WMSServiceException.java

```diff
@@ -0,0 +1,133 @@
+/* (c) 2015 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wms;
+
+import org.geoserver.platform.ServiceException;
+
+/**
+ * Extension of ServiceException for returning a WebMap with the exception.
+ * Used with WMS option EXCEPTIONS=PARTIALMAP to return a partial image if there is an 
+ * exception (such as a timeout) thrown when rendering a WMS request.
+ */
+public class WMSServiceException extends ServiceException {
```

Comment:
Given that this is only intended for exceptions that also return a WebMap, I think refactoring the name would be better; maybe something like WMSPartialMapException. (Since we could get this thrown if we hit other WMS configuration limits such as max request memory or max errors, not just timout).